### PR TITLE
Add drag-and-drop interface & add implementation of DAG running engine for drag-and-drop module

### PR DESCRIPTION
**Pull request by rayrayraykk**
## Checklist

Please check the following items before code is ready to be reviewed.

- [X]  Code has passed all tests
- [X]  Docstrings have been added/updated in Google Style
- [X]  Documentation has been updated
- [X]  Code is ready for review

[Demo](https://img.alicdn.com/imgextra/i3/O1CN01uioF2Z1tYCsWgR3Cf_!!6000000005913-1-tps-2156-1080.gif)

### CODE REVIEW

File: src/agentscope/web/workstation/workflow_utils.py [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+    Args:
+        pipeline (PipelineBase): The pipeline from which to extract agents.
+        seen_agents (set, optional): A set of agents that have already been
+            seen to avoid duplication. Defaults to None.
+
+    Returns:
+        list: A list of unique agent objects found in the pipeline.
+    """
+    if seen_agents is None:
+        seen_agents = set()
```
Comment (pan-x-c) [inline src/agentscope/web/workstation/workflow_utils.py:487]: could use `agent_id` to distinguish different agents

File: src/agentscope/web/README.md [hunk 2]
```diff
@@ -94,4 +94,19 @@ If you implement a video agent, `as_studio` will display the video as follows.
 **Principle:** When calling agent's `self.speak(msg)` method, it will
 output `msg.video_path` to the frontend. Here `msg.video_path` is a list of paths of videos.
 
+## AgentScope Workstation
+A draggable interface for building AgentScope workflow, which is a tool for generating config for running with AgentScope.
+![](https://gw.alicdn.com/imgextra/i3/O1CN01uioF2Z1tYCsWgR3Cf_!!6000000005913-1-tps-2156-1080.gif)
 
+### How to Use
+Go to AgentScope Workstation website, login with GitHub. Drag modules to the convas, and fill the blank. Link each module to build a workflow. Then click export to get the final configurations. Then save the configurations as `config.json`. Then run with the following commands:
```
Comment (DavdGao) [inline src/agentscope/web/README.md:102]: Provide a link here for the web service.

File: src/agentscope/web/README.md [hunk 3]
```diff
@@ -94,4 +94,19 @@ If you implement a video agent, `as_studio` will display the video as follows.
...
+
+```bash
+# Run in command line
+as_workflow config.json
```
Comment (DavdGao) [inline src/agentscope/web/README.md:106]: What's the difference between `as_workflow config.json` and `as_studio config.json`?

Comment (DavdGao) [inline src/agentscope/web/README.md:106]: What's the difference between `as_workflow config.json` and `as_studio config.json`?

**Review by DavdGao — COMMENTED** [review]
plz see inline comments

File: src/agentscope/web/README.md (hunk 3 above)
Comment (DavdGao) [reply src/agentscope/web/README.md:106]: lgtm

**Review by pan-x-c — APPROVED** [review]
LGTM

**Pull request merged**